src/imports.rs

```diff
@@ -250,13 +250,28 @@ impl<'a> FmtVisitor<'a> {
         self.last_pos = pos_after_last_use_item;
     }
 
-    pub fn format_import(&mut self, vis: &ast::Visibility, vp: &ast::ViewPath, span: Span) {
+    pub fn format_import(
+        &mut self,
+        vis: &ast::Visibility,
+        vp: &ast::ViewPath,
+        span: Span,
+        attrs: &[ast::Attribute],
+    ) {
         let vis = utils::format_visibility(vis);
         // 4 = `use `, 1 = `;`
         let rw = Shape::indented(self.block_indent, self.config)
             .offset_left(vis.len() + 4)
             .and_then(|shape| shape.sub_width(1))
-            .and_then(|shape| vp.rewrite(&self.get_context(), shape));
+            .and_then(|shape| match vp.node {
+                // If we have an empty path with attributes attached, we want to skip erasing it
+                ast::ViewPath_::ViewPathList(ref path, ref path_list)
+                    if path_list.is_empty() && !attrs.is_empty() =>
+                {
+                    rewrite_path(&self.get_context(), PathContext::Import, None, path, shape)
```

Comment:
...however, I would structure it just a little bit differently - I would change the Rewrite impl so that it never replaces an empty list with nothing. Then make the check here a positive check - if `path_list.is_empty() && attrs.is_empty()` then `""`, otherwise call `vp.rewrite(...)`. You could also assert in `rewrite` that either the path list or the attributes are non-empty.